docs/user-guide/multicontract/index.md

```diff
@@ -0,0 +1,675 @@
+
+[^reentrancy]: The Prover assumes that the external calls do not modify the
+  storage of the calling contract.  This assumption comes from an assumption
+  that the called code is non-reentrant.  If you are concerned about violations
+  caused by reentrancy, you can override this assumption using a `HAVOC_ALL`
+  summary; see {ref}`havoc-summary` for details.
+
+We can see this behavior by verifying the `integrityOfDeposit` rule against the
+`Pool` contract without giving the Prover access to the `Asset` contract ([full script][havoc-script]):
+
+```sh
+$ certoraRun contracts/Pool.sol --verify Pool:certora/specs/pool_havoc.spec ...
+```
+
+In this case, the `integrityOfDeposit` rule fails.  To understand why, we can
+unfold the call trace for the call to `deposit`:
+
+![Call trace for `integrityOfDeposit` with `deposit` method unfolded to show DEFAULT HAVOCs for calls to `balanceOf` and `transferFrom`](no-link-call-trace.png)
+
+Here we see that the calls to `transferFrom` and `balanceOf` are marked with
+"DEFAULT HAVOC".  This means that the Prover lets the call to
+`transferFrom` to change the balances any way it likes.  In fact, calls to
+`asset.balanceOf(...)` are also unresolved, so the Prover can choose any return
+value that causes a counterexample.  In this case, we can see that the Prover
+chose `3` for the first return value of `balanceOf` and `9`
+for the last return value of `balanceOf`:
+
+![Call trace for `integrityOfDeposit` on `Pool` showing call to `assetBalance` with internal havoced call to `balanceOf`, returning 3 in once place and 9 in another](no-link-variables.png)
+
+The "Call Resolution" tab on the report provides more information about all of
+the unresolved method calls within the contract and how they are resolved by
+the Prover[^resolutionWarnings]:
+
+![Call resolution for `integrityOfDeposit` showing havocs of return values for `balanceOf` and all variables of external contracts for `transferFrom`](no-link-call-resolution.png)
+
+[^resolutionWarnings]: Unresolved calls that are not explicitly handled are
+  considered warnings; in this case there are three unresolved calls, which is
+  why there is a red 3 on the call resolution tab.  In general, it is good
+  practice to explicitly resolve all calls.
+
+Here we see that the call from `Pool.deposit` to `balanceOf` is summarized by
+havocing only the return value (since `balanceOf` is a view method), while the
+call from `Pool.deposit` to `transferFrom` havocs all contracts except `Pool`.
+
+Working with known contracts
+----------------------------
+
+In the case of the `Pool` verification, we don't want the Prover to choose
+arbitrary behavior for the `Asset`, because we have the `Asset` code.  Instead,
+we would like the Prover to model the `asset` contract using the `Asset` code.
+
+To do so, we must first add the `Asset` contract to the set of contracts that
+the Prover knows about.  This set of contracts is called the {term}`scene`.  You
```

Comment:
We should add a note here (or elsewhere) about the `--address` option (maybe a footnote)

Reply:
fixed